### DISCUSSION TITLE
Energy dashboard: Allow addition of unlisted entities (manual entry of sensors)

### DISCUSSION THREAD

**Discussion by AleXSR700** [home-assistant/frontend#17436] [Dashboard]
Hi everyone,

so, I know you are trying to control what the user can enter by prefiltering the avavilable entities.

However, sometimes names change (including entity_id) or a broken device is replaced.
The energy data is then still in the database, but the dashboard can only access them if thy are in the list. 

When the entity is left in the dashboard after removing, the data can still be checked and still goes into the calculations.  With two issues:
1. there is a constant error message that the device is no longer defined.
2. if you delete it, you will never be able to add it again

I always found it a bit of a downside of HA that you cannot "archive" a device/entity so it remains available in HA dropdowns without being shown as "unavailable". And it is a downside that you cannot manually add a device and override the dropdown preselection.

Thank you for your consideration
Alex

**Comment by derjoerg** [comment]
Hi,
same question from my side (here is my usecase):

Till last week I had a simple PV-inverter without the possibility to get the "created solar power" directly from the device. So I installed a Shelly Pro 3EM and all was fine (including energy dashboard).
Since last week I got a new inverter, where I am able to read out all data through ModbusTCP. This is also working fine
My electrician removed the Shelly, as the functionality isn't needed anymore
Now I have the problem with the energy dashboard
HA complains that the Shelly isn't available anymore (which is correct) and that I should remove it
But when I do that I will also loose all history in the energy dashboard regarding the past solar energy 😦

**Comment by derjoerg** [comment]
I found the following in the forum: https://community.home-assistant.io/t/how-to-keep-old-energy-statistics-history-when-changing-the-main-meter-sensor/590042/4

**Reply by AleXSR700** [reply]
Yes, it is possible to keep sensors "alive" by creating fake sensors. However, this takes regular updates and lots of sensors over time (I guess in the last 4 years I have changed around 20 sensors, resulting in loss of data access or constant error messages.

While possible to now use a fake sensor for each, a built-in solution would be much nicer and much better HA quality.

A "simple" archive function in HA where the user can set a device to no longer be updated but remain known. This archived status would allow the user to reassign the same hardware to a new device and would move the archived device to e.g. "disabled devices" in the Devices and Entities pane of HA (or to a new "archived" category).

**Comment by karwosts** [comment]
> When the entity is left in the dashboard after removing, the data can still be checked and still goes into the calculations. With two issues:
> 
>     1. there is a constant error message that the device is no longer defined.
> 
>     2. if you delete it, you will never be able to add it again

I'm actually not sure I agree with this point `2`. I have a sensor that I deleted the entity. I see the "Entity not defined" warning in the energy dashboard configuration. But it is still selectable as a source for my energy grid. It no longer has a friendly name or icon, but the entity_id is still there for picking. 

Are you certain you can't re-add it if it is deleted?
